pkg/build/strategies/sti/sti.go

```diff
@@ -273,7 +274,9 @@ func (builder *STI) Prepare(config *api.Config) error {
 
 	// fetch sources, for their .s2i/bin might contain s2i scripts
 	if len(config.Source) > 0 {
-		if builder.sourceInfo, err = builder.source.Download(config); err != nil {
+		// NOTE: needs timing
+		builder.sourceInfo, err = duration.DownloadTimer(builder.source.Download, builder.result.BuildInfo, config)
```

Comment:
ok, will add just the failure reasons.
because the card (the 3.5 one) says we need the times for assembly and image pull?
https://trello.com/c/KmmYvStu/996-5-separate-out-build-substatus-and-times